daemon/list.go

```diff
@@ -23,35 +21,41 @@ func (daemon *Daemon) List() []*Container {
 	return daemon.containers.List()
 }
 
+type ContainersConfig struct {
+	All     bool
+	Since   string
+	Before  string
+	Limit   int
+	Size    bool
+	Filters string
+}
+
 type ByCreated []types.Container
 
 func (r ByCreated) Len() int           { return len(r) }
 func (r ByCreated) Swap(i, j int)      { r[i], r[j] = r[j], r[i] }
 func (r ByCreated) Less(i, j int) bool { return r[i].Created < r[j].Created }
 
-func (daemon *Daemon) Containers(job *engine.Job) error {
+func (daemon *Daemon) Containers(config *ContainersConfig) ([]types.Container, error) {
```

Comment:
`types.Container` is superbig with slices and maps, makes sense to make it `[]*types.Container`.